src/quoridor/game/game.go

```diff
@@ -10,11 +10,82 @@ import (
 
 // Game is the controller  
 type Game struct {
-	Id    string
+	ID    string `json:"id"`
 	Pawn  Pawn  `json:"pawn"`
+	Fences []Fence `json:"fences"`
 	Board *Board `json:"board"`
 }
 
+//AddFence add the fence on the board
+func (g Game) AddFence(fence Fence) error {
```

Comment:
It's more convenient to use functionnal paradigm instead of oriented object to apply transformation to your game object. This way, you can keep it immutable.

So, your method will be func AddFence(Game game, fence Fence) Game {}, where your output game is a new one (deep-copied)

Same for other methods.